### DISCUSSION TITLE
Allow permanent deletion of templates

### DISCUSSION THREAD

**Discussion by gauravgahlot** [tinkerbell/tink#461] [Ideas]
## Expected Behaviour

A user should be able to permanently delete templates from the database.

## Current Behaviour

Right now a template is only marked as deleted by setting the `deleted_at` field. However, it continues to stay on the database, like forever. Here is a good example of why we need this:

```
tinkerbell=# select count(id) from template where deleted_at IS null;
 count
-------
     9
(1 row)

tinkerbell=# select count(id) from template;
 count
-------
   261
(1 row)
```

Only 9 of 261 templates can be used to create new workflows. While a user may want to keep another 50 templates (maybe because they were used to create workflows), one should be able to delete the rest permanently.

The storage is cheap, but that does not mean we should refrain users from freeing up some space. With the introduction of template revisions, this becomes even more critical. This is because a deleted template may have 30 odd revisions that will eat up the storage forever. 

## Possible Solution

We can use either of the following solutions to permanently delete the templates. Also, we must prompt the user for confirmation.
- Introduce a new flag `-f` (or similar) to the current `delete` command.

```
$ tink template delete -f [id-1] [id-2]...[id-N]
```

- Introduce a new sub-command `purge`. 

```
$ tink template purge [id-1] [id-2]...[id-N]
```

**Comment by gianarb** [comment]
Thank you for taking the time. I will convert it into a discussion. I think we need something similar BUT, when you did this project at the beginning logical deletion vs physical one was a decision, not a mistake I suppose.

Cleaning up the database is for sure important but I want to have a garbage collector that does it automatically even for other resources like workflows execution and events. I don't think we need a command as docker does for example.
